### PR TITLE
[quasimodo] Integrate mongodb world_state observations and enable querying individual databases

### PR DESCRIPTION
**Pull request by nilsbore**
This PR integrates quasimodo with the world_state representation of the object search framework. It enables us to use the mongodb observations intermixed with the meta room ones. Further, it uses this representation to enable @jekekrantz to query for top matches among his built models to speed up matching of models in his database. This will be used to associate observations from @jayyoung 's object search system.

If @jekekrantz could test this before merging, that would be good. Note that when searching for objects in your models, you need to pass the flag `query_msg.query_kind = quasimodo_msgs::retrieval_query::MONGODB_QUERY`.
To add a model to the database, you need to have `mongodb` running, as well as `rosrun quasimodo_object_search insert_object_server.py`. You then need to pass a service according to this definition, in `quasimodo_msgs/insert_model.srv`:

```
uint64 INSERT=1
uint64 REMOVE=2
uint64 action
quasimodo_msgs/model model
string object_id

---
uint64 vocabulary_id
string object_id
```

This is currently announced at `/insert_model_service`. For `INSERT`, you only need to pass the `model` argument. You then get an `object_id` back that you need to keep track of. When you want to remove this model again, pass `REMOVE` together with the `object_id`.

@jayyoung If you have `soma_manager/soma2_local.launch` running together with `quasimodo_object_search/create_object_search_observation.py`, this code will automatically add new entries added to the `world_state` database. For querying these observations, the procedure is the same as above. ATM, you need a `semanticMap` database to train the underlying representation to be able to use this.

### CODE REVIEW

**Comment by nilsbore** [comment]
I also added the ability to query using the incremental version now, simply pass `enable_incremental:=true` to `retrieval.launch`. Atm it's turned off by default since it's not yet integrated with the mongodb queries, but it should be a relatively quick fix.

**Comment by nilsbore** [comment]
Ok, now everything including incremental queries is working also for mongodb results. When we establish e.g. segmentation formats for the observation data base, we need to think about how to represent adjacency of segments so I can use that in the refined segmentation.

**Comment by nilsbore** [comment]
Per @jekekrantz request, I updated the provided service to be equivalent to the topic version.

**Comment by nilsbore** [comment]
Atm, this requires @hkaraoguz https://github.com/hkaraoguz/mongodb_store/tree/messagestorefixcompatibility and https://github.com/hkaraoguz/soma/tree/mongodbcompatibilitychanges, as these are apparently what we are going to use at the deployment.

File: dynamic_object_retrieval/dynamic_object_retrieval/CMakeLists.txt [hunk 1]
```diff
@@ -57,24 +61,23 @@ else()
...
 include_directories(${stopwatch_include_dir})
 
-# Compile the stopwatch project, needed for this
-ExternalProject_Add(stopwatch_project
-  DOWNLOAD_COMMAND ""
-  SOURCE_DIR ${stopwatch_include_dir}
-  BINARY_DIR ${stopwatch_library_dir}
-  BUILD_COMMAND ${CMAKE_MAKE_PROGRAM}
-  INSTALL_COMMAND ""
-)
```
Comment (nilsbore) [inline dynamic_object_retrieval/dynamic_object_retrieval/CMakeLists.txt:76]: @marc-hanheide I don't understand what jenkins is complaining about now. With the project excluded when building with catkin, there should be no builds in the source folder any more. Might there be something lingering on the build server causing the error?

**Comment by marc-hanheide** [comment]
retest this please

**Comment by marc-hanheide** [comment]
yes, you are right. It's a result of destroyed workspaces on several build computers. I hope to have fixed it, but if could show up again for other PRs of `strands_3d_mapping`. If that's the case, notify me.

**Comment by nilsbore** [comment]
Thanks. Haha =) yeah, I guess it might have tried to build this on a few computers by now.

**Comment by nilsbore** [comment]
Ok, now it makes more sense, have to wait for https://github.com/hkaraoguz/soma/tree/mongodbcompatibilitychanges to be merged and released. Ping @hkaraoguz

**Comment by hkaraoguz** [comment]
@nilsbore I have merged it with strands soma update branch.

**Comment by nilsbore** [comment]
Error due to new version of SOMA not being released.

**Comment by nilsbore** [comment]
So this finally(!) passes. @RaresAmbrus OK to merge?

**Comment by nilsbore** [comment]
The failed builds were because of new SOMA not being released btw.

**Comment by nilsbore** [comment]
Ok, @RaresAmbrus , I think this is fine to merge now, I included some licensing text in the stopwatch project.

**Pull request merged**